### PR TITLE
Avoid including hive views in iceberg SHOW TABLES

### PR DESCRIPTION
**Pull request by raunaqmorarka**

### CODE REVIEW

**Review by findepi — COMMENTED** [review]
LGTM

Please add a test exercising Hive and Iceberg using same backing metastore.
This should be a product test to get good coverage of metastore-dependent interactions like this. (For example -- can `getTablesWithParameter` be used with TABLE_COMMENT property?)

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataListing.java [hunk 1]
```diff
@@ -87,14 +93,28 @@ public void setUp()
...
         assertQuerySucceeds("CREATE TABLE iceberg.test_schema.iceberg_table1 (_string VARCHAR, _integer INTEGER)");
         assertQuerySucceeds("CREATE TABLE iceberg.test_schema.iceberg_table2 (_double DOUBLE) WITH (partitioning = ARRAY['_double'])");
+        assertQuerySucceeds("CREATE MATERIALIZED VIEW iceberg.test_schema.iceberg_materialized_view AS " +
+                "SELECT * FROM iceberg.test_schema.iceberg_table1");
+        storageTable = getStorageTable("iceberg", "test_schema", "iceberg_materialized_view");
+
         assertQuerySucceeds("CREATE TABLE hive.test_schema.hive_table (_double DOUBLE)");
-        assertEquals(ImmutableSet.copyOf(metastore.getAllTables("test_schema")), ImmutableSet.of("iceberg_table1", "iceberg_table2", "hive_table"));
+        assertQuerySucceeds("CREATE VIEW hive.test_schema.hive_view AS SELECT * FROM hive.test_schema.hive_table");
+        assertThat(metastore.getAllTables("test_schema"))
```
Comment (sopel39) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataListing.java:102]: move that to actual test method (if we want to keep it)

**Review by sopel39 — APPROVED** [review]

**Comment by raunaqmorarka** [comment]
> LGTM
> 
> Please add a test exercising Hive and Iceberg using same backing metastore.
> This should be a product test to get good coverage of metastore-dependent interactions like this. (For example -- can `getTablesWithParameter` be used with TABLE_COMMENT property?)

Thanks for pointing out, turns out TABLE_PARAMETER_SAFE_VALUE_PATTERN blocks our parameter value, i'll need to tweak that a bit

**Comment by findepi** [comment]
Your value is not far from matching `TABLE_PARAMETER_SAFE_VALUE_PATTERN`

based on the code comment:

>  the HMS may want the parameter
         * value to follow a Java regex pattern or a SQL pattern

seems like we could allow spaces in `TABLE_PARAMETER_SAFE_VALUE_PATTERN`.

**Comment by raunaqmorarka** [comment]
> Your value is not far from matching `TABLE_PARAMETER_SAFE_VALUE_PATTERN`
> 
> based on the code comment:
> 
> > the HMS may want the parameter
> > * value to follow a Java regex pattern or a SQL pattern
> 
> seems like we could allow spaces in `TABLE_PARAMETER_SAFE_VALUE_PATTERN`.

I got things working for iceberg by allowing spaces in TABLE_PARAMETER_SAFE_VALUE_PATTERN.
Also added a product test.
But I also see that in shared metastore setup, hive connector lists all the iceberg tables and MVs with same schema name because it calls `getAllTables`
In shared setup, are we mainly concerned about the iceberg side or the hive side listings also need to work correctly ?

**Review by anjalinorwood — APPROVED** [review]
LGTM!

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataListing.java [hunk 2]
```diff
@@ -103,15 +115,37 @@ public void tearDown()
...
+                        "'" + storageTable.getTableName() + "'");
     }
 
     @Test
     public void testTableColumnListing()
     {
         // Verify information_schema.columns does not include columns from non-Iceberg tables
-        assertQuery("SELECT table_name, column_name FROM iceberg.information_schema.columns WHERE table_schema = 'test_schema'",
-                "VALUES ('iceberg_table1', '_string'), ('iceberg_table1', '_integer'), ('iceberg_table2', '_double')");
+        // TODO this should include columns from the materialized view as well
```
Comment (sopel39) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataListing.java:140]: please create an issue in Github and reference it here

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveMetadataListing.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+                .iterator());
+
+        onTrino().executeQuery("CREATE TABLE hive.default.hive_table (_double DOUBLE)");
+        onTrino().executeQuery("CREATE VIEW hive.default.hive_view AS SELECT * FROM hive.default.hive_table");
+    }
+
+    @AfterTestWithContext
+    public void cleanUp()
+    {
+        onTrino().executeQuery("DROP TABLE hive.default.hive_table");
```
Comment (sopel39) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveMetadataListing.java:53]: drop if exists

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveMetadataListing.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        onTrino().executeQuery("DROP VIEW hive.default.hive_view");
+        onTrino().executeQuery("DROP MATERIALIZED VIEW iceberg.default.iceberg_materialized_view");
+        onTrino().executeQuery("DROP TABLE iceberg.default.iceberg_table1");
+    }
+
+    @Test(groups = {ICEBERG, STORAGE_FORMATS})
+    @Flaky(issue = "https://github.com/trinodb/trino/issues/4864", match = "Failed to read footer of file")
```
Comment (sopel39) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveMetadataListing.java:60]: can this really happen when listing tables and columns?

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveMetadataListing.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+{
+    private String storageTable;
+
+    @BeforeTestWithContext
+    public void setUp()
+    {
+        onTrino().executeQuery("CREATE TABLE iceberg.default.iceberg_table1 (_string VARCHAR, _integer INTEGER)");
+        onTrino().executeQuery("CREATE MATERIALIZED VIEW iceberg.default.iceberg_materialized_view AS " +
+                "SELECT * FROM iceberg.default.iceberg_table1");
+        storageTable = getOnlyElement(onTrino().executeQuery("SHOW TABLES FROM iceberg.default")
```
Comment (sopel39) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveMetadataListing.java:40]: is storage table created immediately?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataListing.java (hunk 2 above)
Comment (raunaqmorarka) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataListing.java:140]: This should get solved by https://github.com/trinodb/trino/pull/8113

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveMetadataListing.java (hunk 5 above)
Comment (raunaqmorarka) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveMetadataListing.java:40]: yes, in iceberg CREATE MV we create the storage table even before storing the MV definition in HMS

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveMetadataListing.java (hunk 4 above)
Comment (raunaqmorarka) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergHiveMetadataListing.java]: that's unnecessary here, removed

**Pull request merged**